jupyter_server/serverapp.py

```diff
@@ -273,7 +273,7 @@ def init_settings(self, jupyter_app, kernel_manager, contents_manager,
             server_root_dir=root_dir,
             jinja2_env=env,
             terminals_available=False,  # Set later if terminals are available
-            serverapp=self
+            serverapp=jupyter_app
```

Comment:
Is this related to the changes for the logger? Just curious trying to understand the logic.

Reply:
Yeah. This was actually a bug that was exposed in PR (line 42). The serverapp was not actually being passed to the extension handlers; rather, the tornado web application was being passed unintentionally.